metricflow/sql/optimizer/required_column_aliases.py

```diff
@@ -127,13 +131,34 @@ def _visit_parents(self, node: SqlPlanNode) -> None:
     def _tag_potential_cte_node(self, table_name: str, column_aliases: Set[str]) -> None:
         """A reference to a SQL table might be a CTE. If so, tag the appropriate aliases in the CTEs."""
         cte_node = self._current_cte_alias_mapping.get_cte_node_for_alias(table_name)
+
         if cte_node is not None:
             self._current_required_column_alias_mapping.add_aliases(cte_node, column_aliases)
             # `visit_cte_node` will handle propagating the required aliases to all CTEs that this CTE node depends on.
             cte_node.accept(self)
 
     def visit_select_statement_node(self, node: SqlSelectStatementNode) -> None:
         """Based on required column aliases for this SELECT, figure out required column aliases in parents."""
+        # If this SELECT node defines any CTEs, it should override ones that were defined in the outer SELECT in case
```

Comment:
Similar question here - why do we expect collisions?

Reply:
Currently, we do not. But once we start adding CTE nodes within `DataflowToSqlQueryPlanConverter`, there's the potential if unique aliases aren't enforced.

Reply:
Would it be simpler to just enforce unique aliases? Or do you think think it would be hard to do that?